Changelog 24.5.0 (#13906)
conda/conda: CHANGELOG.md

[//]: # (current developments)

## 24.5.0 (2024-05-08)

### Enhancements

* Report which `MatchSpec` item caused `Invalid*Spec` exceptions for more informative error messages. (#11203 via #13598)
* MSYS2 packages can now use the upstream installation prefixes. (#13649)

> * Adds a new `reporters` setting for configure output. (#13736)
> * Add support for CEP-15 `base_url` field in `repodata.json`. (#13137 via #13744)
> * In custom channel settings allow specification of channel URLs using a glob-like wildcard pattern, e.g. for user with auth handler plugins. (#13778 via #13779)

Reviewer: ```suggestion
* Add a new `reporters` setting for configure output. (#13736)
* Add support for CEP-15 `base_url` field in `repodata.json`. (#13137 via #13744)
* In custom channel settings, allow specification of channel URLs using a glob-like wildcard pattern, e.g. for user with auth handler plugins. (#13778 via #13779)
```

### Bug fixes

* Fix `conda notices --json` to correctly output JSON. (#13561)

> * Fix prefix replacement for Windows subdir on unixes. (#13689)

Reviewer: ```suggestion
* Fix prefix replacement for Windows subdirectories on Unix systems. (#13689)
```

Author: ```suggestion
* Fix prefix replacement for Windows `subdir` on Unix. (#13689)
```

* Mark `conda.plan._inject_UNLINKLINKTRANSACTION` as pending deprecation. (#12421)
* Mark `conda.plan._plan_from_actions` as pending deprecation. (#12421)
* Mark `conda.plan.add_defaults_to_specs` as pending deprecation. (#12421)
* Mark `conda.plan.add_unlink` as pending deprecation. (#12421)
* Mark `conda.plan.display_actions` as pending deprecation. (#12421)
* Mark `conda.plan.execute_actions` as pending deprecation. (#12421)
* Mark `conda.plan.get_blank_actions` as pending deprecation. (#12421)
* Mark `conda.plan.install_actions` as pending deprecation. (#12421)
* Mark `conda.plan.print_dists` as pending deprecation. (#12421)
* Mark `conda.plan.revert_actions` as pending deprecation. Use `conda.cli.install.revert_actions` instead. (#12421)
* Mark `conda.plan` as an entrypoint as pending deprecation. (#12421)
* Mark `conda.activate._Activator.add_export_unset_vars` as pending deprecation. Use `conda.activate._Activator.get_export_unset_vars` instead. (#13720)
* Mark `conda.activate._Activator.get_scripts_export_unset_vars` as pending deprecation. Use `get_scripts_export_unset_vars` helper function in `test_activate.py` instead. (#13720)

> * Mark `conda.activate._Activator._get_path_dirs(extra_library_bin)` as pending deprecation. Unused argument. (#13720)

Reviewer: ```suggestion
* Mark `conda.activate._Activator._get_path_dirs(extra_library_bin)` as pending deprecation. (#13720)
```

* Mark `conda.activate.JSONFormatMixin.get_scripts_export_unset_vars` as pending deprecation. Use `conda.activate._Activator.get_export_unset_vars` instead. (#13720)
* Mark `conda.gateways.logging.trace` as pending deprecation. Use `Logger.log(conda.common.constants.TRACE, msg)` instead. (#13732)
* Mark `conda create --mkdir` as pending deprecation. The argument is redundant and unnecessary. (#13751)
* Mark `conda install --mkdir` as pending deprecation. Use `conda create` instead. (#13751)
* Mark `conda._vendor.frozendict` as pending deprecation. Use `frozendict` instead. (#13767 via #13766)
* Mark `conda.auxlib.collection.make_immutable` as pending deprecation. Use `frozendict.deepfreeze` instead. (#13801)
* Mark `conda.plan.execute_plan` as pending deprecation. (#13869)
* Mark `conda.plan.execute_instructions` as pending deprecation. (#13869)
* Mark `conda.plan._update_old_plan` as pending deprecation. (#13869)

### Docs

> * Added type hints and doc strings to `conda.core.index`. (#13816)

Reviewer: ```suggestion
* Add type hints and doc strings to `conda.core.index`. (#13816)
```
